lib/client/rpc/market-clients/ammMarkets.ts

```diff
@@ -78,82 +74,106 @@ export class FutarchyAmmMarketsRPCClient implements FutarchyAmmMarketsClient {
-    );
-    const quoteDecimals = unpackMint(ammAcount.quoteMint, quoteInfo).decimals;
-    const quoteAmountCasted = new BN(quoteAmount).mul(
-      new BN(10).pow(new BN(quoteDecimals))
+    const validationError = this.validateAddLiquidity(
+      ammMarket,
+      quoteAmount,
+      maxBasAmount
     );
+    if (validationError) {
+      return validationError;
+    }
 
-    const baseInfo = await this.rpcProvider.connection.getAccountInfo(
-      ammAcount.baseMint
+    const quoteAmountArg = new BN(
+      quoteAmount *
+        new BN(10).pow(new BN(ammMarket.quoteToken.decimals)).toNumber()
     );
-    const baseDecimals = unpackMint(ammAcount.quoteMint, baseInfo);
-    const baseAmountCasted = new BN(baseAmount).mul(
-      new BN(10).pow(new BN(baseDecimals))
+
+    // we just pass this in as min LP tokens and this replicates the calculation in the program
+    const lpTokensToMint = quoteAmountArg
+      .mul(new BN(ammMarket.lpMintSupply))
+      .div(ammMarket.quoteAmount);
+
+    const baseAmountArg = new BN(
+      maxBasAmount *
+        new BN(10).pow(new BN(ammMarket.baseToken.decimals)).toNumber()
     );
 
     const ix = this.ammClient.addLiquidityIx(
-      ammAddr,
-      ammAcount.baseMint,
-      ammAcount.quoteMint,
-      baseAmountCasted,
-      quoteAmountCasted,
-      minLpTokens,
+      ammMarket.publicKey,
+      ammMarket.baseMint,
+      ammMarket.quoteMint,
+      quoteAmountArg,
+      baseAmountArg,
+      lpTokensToMint,
       this.rpcProvider.publicKey
     );
     const tx = await ix.transaction();
     return this.transactionSender.send([tx], this.rpcProvider.connection);
   }
 
   async simulateAddLiquidity(
-    ammAddr: PublicKey,
+    ammMarket: AmmMarket,
     baseAmount: number,
     quoteAmount: number
   ) {
-    const ammAccount = await this.ammClient.getAmm(ammAddr);
-
-    const baseReserves = ammAccount.baseAmount;
-    const quoteReserves = ammAccount.quoteAmount;
-
-    const lpMintInfo = await this.rpcProvider.connection.getAccountInfo(
-      ammAccount.lpMint
-    );
-    const lpMintSupply = Number(
-      unpackMint(ammAccount.lpMint, lpMintInfo).supply
-    );
+    const baseReserves = ammMarket.baseAmount;
+    const quoteReserves = ammMarket.quoteAmount;
 
     const simulation = this.ammClient.simulateAddLiquidity(
       baseReserves,
       quoteReserves,
-      lpMintSupply,
+      ammMarket.lpMintSupply,
       baseAmount,
       quoteAmount
     );
 
     return simulation;
   }
 
-  async removeLiquidity(ammAddr: PublicKey, lpTokensToBurn: number) {
-    const ammAcount = await this.ammClient.getAmm(ammAddr);
-
+  async removeLiquidity(ammMarket: AmmMarket, lpTokensToBurn: number) {
     const minBaseAmount = 23;
     const minQuoteAmount = 20;
     const ix = this.ammClient.removeLiquidityIx(
-      ammAddr,
-      ammAcount.baseMint,
-      ammAcount.quoteMint,
+      ammMarket.publicKey,
+      ammMarket.baseMint,
+      ammMarket.quoteMint,
       lpTokensToBurn,
```

Comment:
This can't be a number it needs to be a BN.

Reply:
ammAddr: PublicKey,
    baseMint: PublicKey,
    quoteMint: PublicKey,
    lpTokensToBurn: BN,
    minBaseAmount: BN,
    minQuoteAmount: BN

Reply:
done now